sql/postgres/inspect.go

```diff
@@ -344,6 +360,77 @@ func (i *inspect) indexes(ctx context.Context, s *schema.Schema) error {
 	return rows.Err()
 }
 
+var reIndexType = regexp.MustCompile("(?i)USING (BTREE|GIN|GIST)")
+
+// indexType extracts index type information from index create statement. see: https://www.cockroachlabs.com/docs/stable/create-index.html
+func indexType(createStmt string) *IndexType {
+	t := reIndexType.FindStringSubmatch(createStmt)
+	return &IndexType{T: t[1]}
+}
+
+func (i *inspect) crdbAddIndexes(s *schema.Schema, rows *sql.Rows) error {
```

Comment:
Please add a test case for this scanning logic. It will be easier to understand this block when looking at the test.